### DISCUSSION TITLE
Questions about the RayTracing - Mobility

### DISCUSSION THREAD

**Discussion by TheoriginalMMM** [NVlabs/sionna#622] [Q&A]
## Question about the RayTracing Module for Mobile Receiver Scenarios

Hello,

I have a few questions regarding the RayTracing module, particularly in scenarios involving a mobile receiver. I've observed two potential approaches to obtain channel measurements corresponding to various user locations:

1. **Single Receiver Approach**: 
   - Add one receiver to the scene.
   - Compute paths and obtain the Channel Impulse Response (CIR).
   - Extract the \(h_f\) corresponding to this specific location.
   - Update the user's location and repeat the last three steps.

2. **Multiple Receivers Approach**:
   - Add \(N\) receivers to the scene at different locations that correspond to the trajectory points we wish to simulate.
   - Compute paths and obtain the CIR.
   - Use the `cir_to_ofdm_channel` function to get the \(h_f\) (CSI).
   - Transpose the resulting tensor to organize the \(h_f\) from different locations in the first dimension.
   - Repeat the previous process if the total number of trajectory points exceeds \(N\).

### Questions:
1. What are the equivalences and differences between these two approaches?
2. Are there any parameter adaptations needed for the second approach?
3. While it's evident that the first approach will take more time due to the need for multiple ray-tracing runs, could this affect the results of the ray tracing in terms of accuracy or other factors? (we consider that all types of paths implemented in the RT)
4. I have been comparing successive CSI measurements recorded at nearby points, specifically those separated by small distances (less than 0.5 m). During this analysis, I observed discontinuities or sudden changes in the channel coefficient magnitudes at certain positions. What could be the potential explanations for these abrupt variations? Additionally, is the RayTracing module designed to produce continuous CSI measurements for receiver locations that are in close proximity? Are there any guarantees or assurances regarding the continuity of these measurements in such scenario?

Thanks for your help with these questions!

**Comment by jhoydis** [comment]
Hi @TheoriginalMMM,

1.) Both approaches lead to identical results. The only thing which matters is the position of a receiver. Note that there can be some random fluctuations in the channel coefficients if diffuse scattering is activated.

2.) No.

3.) The second approach requires more memory. The optimum number of receivers depends on your hardware. Otherwise there should be no difference.


4.) You typically observe sudden changes when you loose (or get additional) propagation paths. For example the LOS component or a strong reflection vanishes. Diffraction ensures in the former case that the fields decreases in a continuous manner. 
Otherwise, the CSI is perfectly continuous with respect to the receiver location (you can see this nicely at the end of the [mobility tutorial](https://nvlabs.github.io/sionna/examples/Sionna_Ray_Tracing_Mobility.html)).

**Reply by TheoriginalMMM** [reply]
Hi @jhoydis,

Thank you for your previous explanations and insights.

I recently attempted to replicate the demonstrations and comparisons from the [mobility tutorial](https://nvlabs.github.io/sionna/examples/Sionna_Ray_Tracing_Mobility.html). I set both `paths.normalize_delays` and the normalization parameter in `cir_to_ofdm_channel` for the two scenarios. While the results of the last figure changed slightly, I found it intriguing. 

0. **Figure 0**: Resulting figure from the tutorial notebook after the previously described modification.
   ![Figure 0](https://github.com/user-attachments/assets/22b54a62-263d-4359-860f-58f2b6e23438)

In addition, I created a personal scene that is larger than the example, featuring more base transmitters and a single moving receiver. I've observed that the differences between the channel coefficients collected using `apply_doppler` and those obtained with a manually moving user vary depending on the transmitters' positions and the starting location of the receiver. Specifically, I simulate a total displacement of approximately 0.5 m, dividing it into 100 small displacements of 0.05 m. 

I have attached two figures illustrating my observations:

1. **Figure 1**: Channel coefficients from the first transmitter, which is closer to the user.
   ![Figure 1](https://github.com/user-attachments/assets/e216360e-6bb3-4dee-8ab9-5455fa637e94)
   
2. **Figure 2**: Channel coefficients from the eighth transmitter, which is farther away.
   ![Figure 2](https://github.com/user-attachments/assets/2734109a-b94f-40cb-a0ef-0d0bfa2b993d)

Both figures correspond to the same starting position of the receiver. 

Could you please provide any insights into the observed patterns? Any recommendations or advice would be greatly appreciated.

Thank you in advance for your help!

**Reply by jhoydis** [reply]
It is difficult to explain what happens in Figure 1 without knowing the details of your setup. I am a bit surprised by these abrupt changes. Could you share some code to reproduce these results? Ideally, using one the scene included with Sionna.
